pi_ldapproxy/proxy.py

```diff
@@ -218,19 +226,25 @@ def handleBeforeForwardRequest(self, request, controls, reply):
         :return:
         """
         if isinstance(request, pureldap.LDAPBindRequest):
-            if self.bound:
-                log.warn('Rejected a second bind request in the same connection')
-                self.send_bind_response((False, 'Reusing connections currently unsupported.'), request, reply)
-                return None
-            elif request.dn == '':
+            if self.received_bind_request:
+                # We have already received a bind request in this connection!
+                if self.factory.allow_connection_reuse:
+                    # We need to reset the state before further processing the request
+                    log.info('Reusing LDAP connection, resetting state ...')
+                    self.reset_state()
+                else:
+                    log.warn('Rejected a second bind request in the same connection')
+                    self.send_bind_response((False, 'Reusing connections currently unsupported.'), request, reply)
+                    return None
+            self.received_bind_request = True
+            if request.dn == '':
```

Comment:
In the previous commit this was only evaluated, when we had NO bind (elif).
Now we check an annonymous connection in any case?

Reply:
The previous commit is not yet merged in here :-/ This will be fun!

Reply:
Na - I think this was my same confusion like my previous comment.